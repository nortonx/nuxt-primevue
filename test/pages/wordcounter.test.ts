import { shallowMount } from "@vue/test-utils";
import Wordcounter from "@/pages/wordcounter.vue";
import { describe, it, expect } from "vitest";

describe("Wordcounter page", () => {
  it("should render correctly and match snapshot", () => {
    const wrapper = shallowMount(Wordcounter);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
