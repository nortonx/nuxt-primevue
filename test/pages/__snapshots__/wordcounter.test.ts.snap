// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Wordcounter page > should render correctly and match snapshot 1`] = `
"<div class="px-4 py-5 md:px-6 lg:px-8">
  <div class="grid">
    <div class="col-12">
      <h1 class="text-4xl font-bold text-center mb-4">Word Counter</h1>
      <p class="text-center text-800"> Enter your text below to count the number of words. </p>
    </div>
    <div class="col-8 col-offset-2">
      <textarea-stub modelvalue="" rows="8" cols="60" auto-resize="" class="w-full border-round shadow-2 p-3 text-800"></textarea-stub>
    </div>
    <div class="col-8 col-offset-2 flex">
      <counter-box-stub counter="0" label="Words"></counter-box-stub>
      <counter-box-stub counter="0" label="Characters"></counter-box-stub>
      <counter-box-stub counter="0" label="Phrases"></counter-box-stub>
      <counter-box-stub counter="0" label="Paragraphs"></counter-box-stub>
    </div>
  </div>
</div>"
`;
